aips/aip-103.md

```diff
@@ -0,0 +1,170 @@
+---
+aip: 10X
+title: Enable interoperability for Federated Keyless Accounts for the same issuer (user-pool/tenant)
+author: Oliver He (oliver.he@aptoslabs.com)
+Status: Draft # | Last Call | Accepted | Final | Rejected>
+last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
+type: <Standard (Core, Networking, Interface, Application, Framework) | Informational | Process>
+created: 11/08/2024
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-103.md
+---
+
+# AIP-10X - Enable interoperability for Federated Keyless Accounts for the same issuer (user-pool/tenant)
+
+## Summary
+
+This AIP proposes enabling Aptos Federated Keyless to be interoperable with dApps from the same issuer (user-pool/tenant).
+
+For IAM providers like Auth0 and Cognito, JWT tokens are scoped to a user-pool/tenant via the `iss` field, and they are also scoped to a specific application via the `aud` field.  This means that JWTs from the same issuer but with different `aud` values are from different applications and cannot be used to derive the same Aptos Federated Keyless Account even though they represent the same user identity within the same user-pool/tenant.
+
+Many customers of Auth0 and Cognito have applications with different branding within the same user-pool/tenant ecosystem.  Thus it is natural for such customers to use different application identifiers for their applications for organizational purposes.  This AIP will enable Aptos Federated Keyless Accounts to be interoperable across such applications.
+
+## Impact and risks
+
+- The relaxation of the `aud` field will allow for broader interoperability across applications within the same user-pool/tenant.  This allows for broader adoption of Aptos Federated Keyless Accounts in such user ecosystems.
+
+Risks
+- Developers need to not use `aud`-less accounts when not appropriate. This can be mitigated by the Aptos SDK default behavior so that developers must explicitly enable using `aud`-less accounts.
```

Comment:
"Developers need to not use `aud`-less accounts when not appropriate" is not helpful.

You have to describe when it is and when it isn't appropriate.